File: app/src/cs/emailReply.cs
using System;
using System.IO;
using Newtonsoft.Json.Linq;
using Outlook = Microsoft.Office.Interop.Outlook;

class Program
{
    static void Main(string[] args)
    {
        string BASE_PATH;

        if (args.Length > 0 && Directory.Exists(args[0]))
            BASE_PATH = args[0];
        else
            BASE_PATH = AppDomain.CurrentDomain.BaseDirectory;

        try
        {
            string jsonText = File.ReadAllText(Util.JsonPath);
            JObject json = JObject.Parse(jsonText);

            string clientEmail = (string)json["form"]["clientEmail"];
            string clientLanguage = (string)json["form"]["clientLanguage"];
            string lawyerName = (string)json["lawyer"]["name"];
            string lawyerId = (string)json["lawyer"]["id"];

            // === Load the appropriate HTML template ===
            string templateFile = clientLanguage == "English" ? "en/Reply.html" : "fr/Reply.html";
            string templatePath = Path.Combine(Util.RootDir, "app", "templates", templateFile);
            string htmlBody = File.ReadAllText(templatePath);

            // === Replace placeholder ===
            string lawyerString = Util.GetLawyerString(lawyerName, lawyerId);

            htmlBody = htmlBody.Replace("{{lawyerName}}", lawyerString);

            // === Start Outlook and create mail using Interop ===
            var outlookApp = new Microsoft.Office.Interop.Outlook.Application();
            var mail = (Microsoft.Office.Interop.Outlook.MailItem)outlookApp.CreateItem(Microsoft.Office.Interop.Outlook.OlItemType.olMailItem);

            mail.To = clientEmail;
            mail.Subject = clientLanguage == "English" ? "Reply - Allen Madelin" : "Réponse - Allen Madelin";
            mail.HTMLBody = htmlBody;
            mail.Display();
            WindowFocus.ShowWithFocus(mail.GetInspector);
        }
        catch (Exception ex)
        {
            Console.WriteLine("Error: " + ex.Message);
        }
    }
}

File: app/src/emailReply.py
from office_utils import *
from parse_json import *
import win32com.client as COM

def draft_reply():
    try:
        data = read_json()
        form, _, lawyer = split_data(data)

        client_email = form["clientEmail"]
        client_language = get_language(data)
        lawyer_name = lawyer.get("name", "")
        lawyer_id = lawyer.get("id", "")
        lang = "fr" if client_language.startswith("fr") else "en"

        template_path = os.path.join(TEMPLATES_DIR, lang, "Reply.html")

        with open(template_path, "r", encoding="utf-8") as f:
            html_body = f.read()

        html_body = html_body.replace("{{lawyerName}}", get_lawyer_string(lawyer_name, lawyer_id))

        outlook = COM.Dispatch("Outlook.Application")
        mail = outlook.CreateItem(0)  # 0 = olMailItem
        mail.To = client_email
        mail.Subject = "Réponse - Allen Madelin" if lang == "fr" else "Reply - Allen Madelin"
        mail.HTMLBody = html_body
        mail.Display()
        
        focus_office_window(mail)
        print("Reply email draft opened in Outlook.")

    except Exception as e:
        alert_error(f"Error: {e}")
        raise

if __name__ == "__main__":
    draft_reply()
